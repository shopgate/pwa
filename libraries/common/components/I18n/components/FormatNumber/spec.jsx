import React from 'react';
import { mount } from 'enzyme';
import I18n from '../../index';
import FormatNumber from './index';

describe('<FormatNumber>', () => {
  const pairs = [
    [1, '1.00', 2],
    [0.1, '0.10', 2],
    [1, '1.000', 3],
    [1, '1', 0],
  ];

  /**
   * Makes a component.
   * @param {number} number Number
   * @param {number} fractions Decimal points.
   * @returns {Object}
   */
  const makeComponent = (number, fractions) => (mount((
    <I18n.Provider lang="en-US" locales={{}}>
      <div>
        <FormatNumber number={number} fractions={fractions} />
      </div>
    </I18n.Provider>
  )));

  pairs.forEach(([input, expexted, fractions]) => {
    it(`should format ${input} into ${expexted}`, () => {
      const component = makeComponent(input, fractions);
      expect(component.text().includes(`${expexted}`)).toBe(true);
      expect(component).toMatchSnapshot();
    });
  });

  it('should return same number and className when no context is available', () => {
    const component = mount(<FormatNumber number={1} fractions={2} className="some-class" />);
    expect(component.html().includes('some-class')).toBe(true);
    expect(component.text().includes('1')).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
