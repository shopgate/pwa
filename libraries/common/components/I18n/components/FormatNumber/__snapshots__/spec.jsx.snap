// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatNumber> should format 0.1 into 0.10 1`] = `
<I18nProvider
  lang="en-US"
  locales={Object {}}
>
  <div>
    <FormatNumber
      className=""
      fractions={2}
      number={0.1}
    >
      <span
        className=""
      >
        0.10
      </span>
    </FormatNumber>
  </div>
</I18nProvider>
`;

exports[`<FormatNumber> should format 1 into 1 1`] = `
<I18nProvider
  lang="en-US"
  locales={Object {}}
>
  <div>
    <FormatNumber
      className=""
      fractions={0}
      number={1}
    >
      <span
        className=""
      >
        1
      </span>
    </FormatNumber>
  </div>
</I18nProvider>
`;

exports[`<FormatNumber> should format 1 into 1.00 1`] = `
<I18nProvider
  lang="en-US"
  locales={Object {}}
>
  <div>
    <FormatNumber
      className=""
      fractions={2}
      number={1}
    >
      <span
        className=""
      >
        1.00
      </span>
    </FormatNumber>
  </div>
</I18nProvider>
`;

exports[`<FormatNumber> should format 1 into 1.000 1`] = `
<I18nProvider
  lang="en-US"
  locales={Object {}}
>
  <div>
    <FormatNumber
      className=""
      fractions={3}
      number={1}
    >
      <span
        className=""
      >
        1.000
      </span>
    </FormatNumber>
  </div>
</I18nProvider>
`;

exports[`<FormatNumber> should return same number and className when no context is available 1`] = `
<FormatNumber
  className="some-class"
  fractions={2}
  number={1}
>
  <span
    className="some-class"
  >
    1
  </span>
</FormatNumber>
`;
